VDA-StorageNode-Registry/test/index.test.ts

```diff
@@ -0,0 +1,939 @@
+                    await expect(
+                        contract.addNode(nodeInfo, "0x00", "0x00", "0x00")
+                    ).to.be.revertedWithCustomError(contract, "InvalidLatitude")
+                }
+            })
+
+            it("Failed: Invalid Longitude",async () => {
+                const invalidLongValues = [-180.1, -270, -400.2523, 181, 360, 500.235];
+                for (let i = 0; i < invalidLongValues.length; i++) {
+                    const nodeInfo = createStorageNodeStruct(
+                        didAddress,
+                        "https://1",
+                        "US",
+                        "North America",
+                        datacenterIds[0],
+                        0,
+                        invalidLongValues[i]);
+                    await expect(
+                        contract.addNode(nodeInfo, "0x00", "0x00", "0x00")
+                    ).to.be.revertedWithCustomError(contract, "InvalidLongitude")
+                }
+            })
+
+            it("Failed: No trusted signer",async () => {
+                const nonce = await contract.nonce(user.address);
+                
+                const { requestSignature, requestProof, authSignature } = getAddNodeSignatures(storageNode, nonce, user, trustedSigner);
+
+                await expect(
+                    contract.addNode(storageNode, requestSignature, requestProof, authSignature)
+                ).to.be.revertedWithCustomError(contract, "NoSigners");
+            })
+
+            it("Failed: Invalid auth signature",async () => {
+                await contract.addTrustedSigner(trustedSigner.address);
+
+                const badSigner = Wallet.createRandom();
+
+                const nonce = await contract.nonce(user.address);
+
+                const { requestSignature, requestProof, authSignature } = getAddNodeSignatures(storageNode, nonce, user, badSigner);
+
+                await expect(
+                    contract.addNode(storageNode, requestSignature, requestProof, authSignature)
+                ).to.be.revertedWithCustomError(contract, "InvalidSignature");
+            })
+
+            it("Success", async () => {
+                const nonce = await contract.nonce(user.address);
+
+                const { requestSignature, requestProof, authSignature } = getAddNodeSignatures(storageNode, nonce, user, trustedSigner);
+
+                const tx = await contract.addNode(storageNode, requestSignature, requestProof, authSignature);
+
+                await expect(tx).to.emit(contract, "AddNode").withArgs(
+                    storageNode.didAddress,
+                    storageNode.endpointUri,
+                    storageNode.countryCode,
+                    storageNode.regionCode,
+                    storageNode.datacenterId,
+                    storageNode.lat,
+                    storageNode.long
+                );
+            })
+
+            it("Failed: Registered didAddress & endpointURI", async () => {
```

Comment:
Please rename this test or add more detail, I'm unclear what is eing tested.

Reply:
Updated to `Duplicated didAddress and enpointURI`.
Registered `didAddress` or `endpointURI` can't be duplicated.